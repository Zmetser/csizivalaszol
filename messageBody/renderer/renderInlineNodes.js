/**
 * @flow
 */

import * as React from 'react'
import { UnknownNodeError } from '../errors'

import type { InlineNode } from '../types'

import {
  Text,
  Link
} from '../components'

function createInlineNode (node: InlineNode, index): React.Element<*> {
  switch (node.type) {
    case 'Text': return Text(node, index)
    case 'Link': return <Link {...node} key={index} />
    default: throw new UnknownNodeError(node)
  }
}
export default function renderInlineNodes (nodes: Array<InlineNode>): Array<React.Element<*>> {
  return nodes.map(createInlineNode)
}
