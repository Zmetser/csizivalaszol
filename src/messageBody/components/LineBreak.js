/**
 * Create a br
 */

import * as React from 'react'

export default function LineBreak (): React.Element<'br'> {
  return <br />
}
