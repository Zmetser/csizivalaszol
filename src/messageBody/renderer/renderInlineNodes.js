/**
 * @flow
 */

import * as React from 'react'
import { UnknownNodeError } from '../errors'

import type { InlineNode } from '../types'

import {
  Text,
  Link,
  LineBreak
} from '../components'

function createInlineNode (node: InlineNode, index): React.Element<*> {
  switch (node.type) {
    case 'Text': return <Text node={node} key={index} />
    case 'Link': return <Link node={node} key={index} />
    case 'LineBreak': return <LineBreak key={index} />
    default: throw new UnknownNodeError(node)
  }
}
export default function renderInlineNodes (nodes: Array<InlineNode>): Array<React.Element<*>> {
  return nodes.map(createInlineNode)
}
