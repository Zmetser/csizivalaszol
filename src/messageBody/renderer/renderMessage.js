import * as React from 'react'
import {
  Paragraph,
  List
} from '../components'

import type { BlockNode } from '../types.js'
import { UnknownNodeError } from '../errors'

export default function render (message: Array<BlockNode>): Array<React.Element<*>> {
  return message.map((node, index): React.Element<*> => {
    switch (node.type) {
      case 'Paragraph': return <Paragraph node={node} key={index} />
      case 'List': return <List node={node} key={index} />
      default: throw new UnknownNodeError(node)
    }
  })
}
