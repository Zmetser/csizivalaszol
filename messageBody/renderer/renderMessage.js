/**
 * @flow
 */

import * as React from 'react'
import { Paragraph } from '../components'

import type { BlockNode } from '../types.js'
import { UnknownNodeError } from '../errors'

export default function render (message: Array<BlockNode>): Array<React.Element<*>> {
  return message.map((node, index): React.Element<*> => {
    switch (node.type) {
      case 'Paragraph': return Paragraph({ value: node.value }, index)
      default: throw new UnknownNodeError(node)
    }
  })
}
