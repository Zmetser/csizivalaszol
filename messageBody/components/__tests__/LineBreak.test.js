/**
 * @flow
 */
import renderer from 'react-test-renderer'

import LineBreak from '../LineBreak'

describe('<br />', () => {
  test('line break', () => {
    const element = renderer.create(
      LineBreak()
    )
    expect(element.toJSON()).toMatchSnapshot()
  })
})
