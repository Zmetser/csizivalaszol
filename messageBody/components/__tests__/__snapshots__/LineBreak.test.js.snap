// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<br /> line break 1`] = `<br />`;
